sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -2313,6 +2313,33 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-40988: Test case for insert partition should verify value") {
+    val tableName = "partition_table"
+    withTable(tableName) {
+      sql(s"DROP TABLE IF EXISTS $tableName")
+      sql(s"CREATE TABLE $tableName (id INT, name STRING) USING PARQUET PARTITIONED BY (age INT)")
+
+      sql(s"INSERT INTO TABLE $tableName PARTITION(age=1) VALUES (1, 'ABC')")
+      checkAnswer(spark.table(s"$tableName"), Row(1, "ABC", 1))
+
+      checkError(
+        exception = intercept[SparkNumberFormatException] {
+          sql(s"INSERT INTO TABLE $tableName PARTITION(age='aaa') VALUES (1, 'ABC')")
```

Comment:
Seems this command would fail even before. I guess you want to test insert into hive table ?
```scala
sql(s"CREATE TABLE $tableName (id INT, name STRING) STORED PARQUET PARTITIONED BY (age INT)")

withSQLConf(HiveUtils.CONVERT_INSERTING_PARTITIONED_TABLE.key -> "false") {
  ...
}
```

Reply:
@ulysses-you 

Yes, you are right. I am adding a test case for the insert into the hive table with partitioned data.

The following implementation is correct as per your suggestion?

```scala
      withSQLConf("spark.sql.hive.convertInsertingPartitionedTable" -> "false") {
        checkError(
          exception = intercept[SparkNumberFormatException] {
            sql(s"INSERT INTO TABLE $tableName PARTITION(age='aaa') VALUES (1, 'ABC')")
          },
          errorClass = "CAST_INVALID_INPUT",
          parameters = Map(
            "ansiConfig" -> "\"spark.sql.ansi.enabled\"",
            "expression" -> "'aaa'",
            "sourceType" -> "\"STRING\"",
            "targetType" -> "\"INT\""),
          context = ExpectedContext(
            fragment = s"INSERT INTO TABLE $tableName PARTITION(age='aaa')",
            start = 0,
            stop = 38 + tableName.length)
        )
      }```

Reply:
yeah, can you send a new pr without irrelevant git commit history ? thank you

Reply:
Hi @ulysses-you 

Please review the latest changes.